### PR TITLE
chore(css-vars): refactor css var

### PR DESCRIPTION
**Pull request by TimKolberger**
## 📝 Description

This PR includes a refactoring of `css-var.ts`, adds a helper function for creating and calculating css vars.

## 🚀 New behavior

The creation of css vars for a theme follows this pseudo code:

```ts
function createThemeVars(theme) {
  const tokens = extractTokens(theme)
  const cssVars = {}
  const cssMap = {}

  walkObject(tokens, (value, path) => {
    const key = path.join("-")
    cssVars[key] = value
  })

  return { cssVars, cssMap }
}
```

Create CSS vars:

```ts
const sliderWidth = cssVar("slider-width", "<fallback value>")
const { var, varRef } = sliderWidth
// var = --slider-width
// varRef = var(--slider-width, <fallback value>)
```

And calc CSS vars:

```ts
const translateX = calc.subtract(width, height).multiply(-1).toString()
```

## 💣 Is this a breaking change (Yes/No):

No

### CODE REVIEW

File: packages/styled-system/src/create-theme-vars/create-theme-vars.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      },
+    }
+  },
+  defaultHandler: (keys, value, options) => {
+    const lookupKey = keys.join(".")
+    const varKey = keys.join("-")
+
+    const { variable, reference } = cssVar(
+      varKey,
+      undefined, // TODO should we add a fallback value?
```
Comment (TimKolberger) [inline packages/styled-system/src/create-theme-vars/create-theme-vars.ts:80]: Should we add a fallback value for the css vars?

Comment (segunadebayo) [reply packages/styled-system/src/create-theme-vars/create-theme-vars.ts:80]: Yes, let's add a fallback value.

Comment (segunadebayo) [reply packages/styled-system/src/create-theme-vars/create-theme-vars.ts:80]: I don't think we need to.

Comment (TimKolberger) [reply packages/styled-system/src/create-theme-vars/create-theme-vars.ts:80]: Ok 👍  I'll remove the TODO comments

**Review by segunadebayo — APPROVED** [review]
Looks great, Tim!

Thanks for working on this!

**Pull request merged**